app/queries/admin/moderators_query.rb

```diff
@@ -4,29 +4,32 @@ class ModeratorsQuery
       state: :trusted
     }.with_indifferent_access.freeze
 
+    VALID_ROLES = %i[banned warned trusted comment_banned].freeze
+
     def self.call(relation: User.all, options: {})
       options = DEFAULT_OPTIONS.merge(options)
       state, search = options.values_at(:state, :search)
+      role_id = Role.find_by(name: state)&.id
 
       relation = if state.to_s == "potential"
                    relation.where(
-                     "id NOT IN (SELECT user_id FROM users_roles WHERE role_id = ?)
-                     AND id NOT IN (SELECT user_id FROM users_roles WHERE role_id = ?)",
-                     role_id_for(:trusted),
-                     role_id_for(:banned),
+                     "id NOT IN (SELECT user_id FROM users_roles WHERE role_id IN (?))",
+                     potential_role_ids,
                    ).order("users.comments_count" => :desc)
-                 else
+                 elsif role_id.present?
                    relation.joins(:roles)
-                     .where(users_roles: { role_id: role_id_for(state) })
+                     .where(users_roles: { role_id: role_id })
+                 else
+                   User.none
```

Comment:
This seems like an over-optimization to me but 🤷‍♂️

Reply:
IMHO this is great 👍 Consistently returning AR collections is a good practice that we should follow more.